src/utilities/credentials/credentials.ts

```diff
@@ -59,28 +68,70 @@ export async function saveCredential(credential: Credential): Promise<void> {
   await saveList(list);
 }
 
+interface LegacySporranCredential
+  extends Omit<SporranCredential, 'credential'> {
+  request: ICredential & { claimerSignature?: unknown };
+}
+
+function isLegacySporranCredential(
+  input: unknown,
+): input is LegacySporranCredential {
+  return typeof input === 'object' && input !== null && 'request' in input;
+}
+
+function isLegacyICredential(input: unknown): input is LegacyICredential {
+  return (
+    typeof input === 'object' && input !== null && 'claimerSignature' in input
+  );
+}
+
 // SDK <0.29 had claimerSignature in ICredential
 interface LegacyICredential extends ICredential {
   claimerSignature?: unknown;
 }
 
```

Comment:
Would it be nicer to have the interface above the type guard?

Reply:
That’s what I thought as well after creating the PR :) Okay, let me override the laziness and change that!